pyproject.toml

```diff
@@ -17,7 +17,7 @@ classifiers = [
 requires-python = ">=3.10"
 dependencies = [
     "flax~=0.7.0",
-    "jax~=0.4.13",
+    "jax~=0.4.16",
```

Comment:
Hm, I didn't get a deprecation warning weirdly, thanks for mentioning that! I've pushed a change that fixes the type annotations. Out of curiosity, are there any other warnings you get?

Reply:
I actually ended up downgrading the requirement to 0.4.11 since that's the latest version supported by jax-metal (which is a lot faster at least for me on an M1 chip)

Reply:
There is deprecation warning on jax.core.Shape as well (didn't see it at first), this is with using 0.4.18 and can't see this mentioned Jax's changelog. All warnings:
```
=============================== warnings summary ===============================
../cupbearer-env/lib/python3.10/site-packages/flax/linen/module.py:77
  /mimer/NOBACKUP/groups/ml-safety/vikren/mad/cupbearer-env/lib/python3.10/site-packages/flax/linen/module.py:77: DeprecationWarning: jax.random.KeyArray is deprecated. Use jax.Array for ann
otations, and jax.dtypes.issubdtype(arr.dtype, jax.dtypes.prng_key) for runtime detection of typed prng keys.
    KeyArray = Union[jax.Array, jax.random.KeyArray]  # pylint: disable=invalid-name

../cupbearer-env/lib/python3.10/site-packages/flax/linen/recurrent.py:45
  /mimer/NOBACKUP/groups/ml-safety/vikren/mad/cupbearer-env/lib/python3.10/site-packages/flax/linen/recurrent.py:45: DeprecationWarning: jax.random.KeyArray is deprecated. Use jax.Array for 
annotations, and jax.dtypes.issubdtype(arr.dtype, jax.dtypes.prng_key) for runtime detection of typed prng keys.
    PRNGKey = jax.random.KeyArray

../cupbearer-env/lib/python3.10/site-packages/flax/linen/recurrent.py:753
  /mimer/NOBACKUP/groups/ml-safety/vikren/mad/cupbearer-env/lib/python3.10/site-packages/flax/linen/recurrent.py:753: DeprecationWarning: jax.random.KeyArray is deprecated. Use jax.Array for
 annotations, and jax.dtypes.issubdtype(arr.dtype, jax.dtypes.prng_key) for runtime detection of typed prng keys.
    init_key: Optional[random.KeyArray] = None,

../cupbearer-env/lib/python3.10/site-packages/flax/linen/recurrent.py:979
  /mimer/NOBACKUP/groups/ml-safety/vikren/mad/cupbearer-env/lib/python3.10/site-packages/flax/linen/recurrent.py:979: DeprecationWarning: jax.random.KeyArray is deprecated. Use jax.Array for
 annotations, and jax.dtypes.issubdtype(arr.dtype, jax.dtypes.prng_key) for runtime detection of typed prng keys.
    init_key: Optional[random.KeyArray] = None,

../cupbearer-env/lib/python3.10/site-packages/flax/linen/recurrent.py:1003
  /mimer/NOBACKUP/groups/ml-safety/vikren/mad/cupbearer-env/lib/python3.10/site-packages/flax/linen/recurrent.py:1003: DeprecationWarning: jax.random.KeyArray is deprecated. Use jax.Array fo
r annotations, and jax.dtypes.issubdtype(arr.dtype, jax.dtypes.prng_key) for runtime detection of typed prng keys.
    init_key: Optional[random.KeyArray] = None,

../cupbearer-env/lib/python3.10/site-packages/flax/linen/stochastic.py:28
  /mimer/NOBACKUP/groups/ml-safety/vikren/mad/cupbearer-env/lib/python3.10/site-packages/flax/linen/stochastic.py:28: DeprecationWarning: jax.random.KeyArray is deprecated. Use jax.Array for annotations, and jax.dtypes.issubdtype(arr.dtype, jax.dtypes.prng_key) for runtime detection of typed prng keys.
    KeyArray = Union[jax.Array, jax.random.KeyArray]

../cupbearer-env/lib/python3.10/site-packages/chex/_src/pytypes.py:53
  /mimer/NOBACKUP/groups/ml-safety/vikren/mad/cupbearer-env/lib/python3.10/site-packages/chex/_src/pytypes.py:53: DeprecationWarning: jax.core.Shape is deprecated. Use Shape = Sequence[int | Any].
    Shape = jax.core.Shape

../cupbearer-env/lib/python3.10/site-packages/cupbearer/models/computations.py:239
  /mimer/NOBACKUP/groups/ml-safety/vikren/mad/cupbearer-env/lib/python3.10/site-packages/cupbearer/models/computations.py:239: DeprecationWarning: jax.random.KeyArray is deprecated. Use jax.Array for annotations, and jax.dtypes.issubdtype(arr.dtype, jax.dtypes.prng_key) for runtime detection of typed prng keys.
    key: jax.random.KeyArray, shape: Sequence[int], dtype=jnp.float_, value=0.0

../cupbearer-env/lib/python3.10/site-packages/cupbearer/models/computations.py:245
  /mimer/NOBACKUP/groups/ml-safety/vikren/mad/cupbearer-env/lib/python3.10/site-packages/cupbearer/models/computations.py:245: DeprecationWarning: jax.random.KeyArray is deprecated. Use jax.Array for annotations, and jax.dtypes.issubdtype(arr.dtype, jax.dtypes.prng_key) for runtime detection of typed prng keys.
    key: jax.random.KeyArray, shape: Sequence[int], dtype=jnp.float_

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
```

Reply:
Thanks, not sure why I'm not seeing these. I generally dislike the many warnings a lot of ML code produces and would like to avoid them if possible. Seems like I do see warnings in the pytorch refactor though, so probably won't look into this (and just fix the pytorch ones where possible)